### PR TITLE
Convert hui-view to Lit

### PR DESCRIPTION
**Pull request by balloob**
Converts hui-view to Lit.

### CODE REVIEW

File: src/panels/lovelace/hui-view.ts [hunk 1]
```diff
@@ -1,20 +1,73 @@
...
+    };
+  }
+
+  constructor() {
+    super();
+    this._cards = [];
+    this._badges = [];
+  }
+
+  protected render() {
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
  protected render(): TemplateResult {
```

File: src/panels/lovelace/hui-view.ts [hunk 2]
```diff
@@ -1,20 +1,73 @@
...
+                }"
+                @click="${this._addCard}"
+              ></paper-fab>
+            `
+          : ""
+      }
+    `;
+  }
+
+  protected renderStyles() {
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
  protected renderStyles(): TemplateResult {
```

File: src/panels/lovelace/hui-view.ts [hunk 3]
```diff
@@ -79,58 +132,60 @@ class HUIView extends localizeMixin(EventsMixin(PolymerElement)) {
...
+      this._createCards(lovelace.config.views[this.index!]);
+    } else if (changedProperties.has("hass")) {
+      this._cards.forEach((element) => {
+        element.hass = this.hass;
+      });
+    }
   }
 
-  _addCard() {
+  private _addCard() {
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
  private _addCard(): void {
```

File: src/panels/lovelace/hui-view.ts [hunk 4]
```diff
@@ -79,58 +132,60 @@ class HUIView extends localizeMixin(EventsMixin(PolymerElement)) {
...
-      lovelace: this.lovelace,
-      path: [this.index],
+      lovelace: this.lovelace!,
+      path: [this.index!],
     });
   }
 
-  _createBadges(config) {
-    const root = this.$.badges;
+  private _createBadges(config: LovelaceViewConfig) {
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
  private _createBadges(config: LovelaceViewConfig): void {
```

File: src/panels/lovelace/hui-view.ts [hunk 5]
```diff
@@ -157,8 +216,8 @@ class HUIView extends localizeMixin(EventsMixin(PolymerElement)) {
     root.style.display = elements.length > 0 ? "block" : "none";
   }
 
-  _createCards(config) {
-    const root = this.$.columns;
+  private _createCards(config: LovelaceViewConfig) {
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
  private _createCards(config: LovelaceViewConfig): void {
```

File: src/panels/lovelace/hui-view.ts [hunk 6]
```diff
@@ -233,28 +292,14 @@ class HUIView extends localizeMixin(EventsMixin(PolymerElement)) {
...
-  _lovelaceChanged(lovelace) {
-    if (lovelace.editMode && !editCodeLoaded) {
-      editCodeLoaded = true;
-      import(/* webpackChunkName: "hui-view-editable" */ "./hui-view-editable");
-    }
+declare global {
+  interface HTMLElementTagNameMap {
+    "hui-view": HUIView;
   }
 }
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
customElements.define("hui-view", HUIView);
```

File: src/panels/lovelace/hui-view.ts [hunk 7]
```diff
@@ -1,20 +1,73 @@
...
+      ${this.renderStyles()}
+      <div id="badges"></div>
+      <div id="columns"></div>
+      ${
+        this.lovelace!.editMode
+          ? html`
+              <paper-fab
+                elevated="2"
+                icon="hass:plus"
+                title="${
```
Comment (zsarnett) [inline src/panels/lovelace/hui-view.ts]: `.title`?

File: src/panels/lovelace/hui-view.ts [hunk 8]
```diff
@@ -1,20 +1,73 @@
...
+    return html`
+      ${this.renderStyles()}
+      <div id="badges"></div>
+      <div id="columns"></div>
+      ${
+        this.lovelace!.editMode
+          ? html`
+              <paper-fab
+                elevated="2"
+                icon="hass:plus"
```
Comment (iantrich) [inline src/panels/lovelace/hui-view.ts]:
```suggestion
                .icon="hass:plus"
```

Comment (iantrich) [reply src/panels/lovelace/hui-view.ts]: 👍

File: src/panels/lovelace/hui-view.ts (hunk 6 above)
Comment (balloob) [reply src/panels/lovelace/hui-view.ts]: It was not in the diff because it didn't change 👍

**Pull request merged**